### PR TITLE
libusb: add version 1.0.24

### PR DESCRIPTION
**Pull request by yipdw**
Specify library name and version:  **libusb/1.0.24**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.  **Tested with Conan 1.32.1.**
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated. **Built on x86-64 Linux with gcc 9.3.0.**

### CODE REVIEW

**Comment by yipdw** [comment]
libusb 1.0.24 changed some things in their Visual Studio support to build under Visual Studio 2019; it looks like some `replace_in_file`s aren't happy about that.  I'll look at that sometime later today.

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — DISMISSED** [review]

**Review by mathbunnyru — APPROVED** [review]

File: recipes/libusb/all/conanfile.py [hunk 1]
```diff
@@ -68,7 +68,13 @@ def system_requirements(self):
 
     def _build_visual_studio(self):
         with tools.chdir(self._source_subfolder):
-            solution_file = "libusb_2017.sln"
+            if tools.Version(self.version) >= "1.0.24":
+                solution_file = "libusb_2019.sln"
+            else:
+                solution_file = "libusb_2017.sln"
+
```
[the remark this answers is not in the document]
Comment (yipdw) [reply recipes/libusb/all/conanfile.py:75]: Thanks, I'll take a look at that; I would like to find a less repetitive way to express this cascade.  A dict is appealing, though the availability of the VS 2019 project file being limited to libusb >= 1.0.24 adds a complication.

[the remark this answers is not in the document]
Comment (mathbunnyru) [reply recipes/libusb/all/conanfile.py:75]:
```
solution_msvc_year = 2019 if tools.Version(self.version) >= "1.0.24" else 2017

solution_msvc_year = {
    "11": 2012,
    "12": 2013,
    "14": 2015,
    "15": 2017
}.get(str(self.settings.compiler.version), solution_msvc_year)

solution_file = "libusb_{}.sln".format(solution_msvc_year)
```

Something like this should do the same thing as you do.

[the remark this answers is not in the document]
Comment (yipdw) [reply recipes/libusb/all/conanfile.py]: Ah, thanks for the example -- I've integrated this in 19b98358e6689863c749032ae31febac697c29f5.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Pull request merged**